server/src/main/java/org/elasticsearch/action/support/tasks/TransportTasksAction.java

```diff
@@ -323,6 +328,9 @@ class NodeTransportHandler implements TransportRequestHandler<NodeTaskRequest> {
 
         @Override
         public void messageReceived(final NodeTaskRequest request, final TransportChannel channel, Task task) throws Exception {
+            if (task instanceof CancellableTask cancellableTask) {
+                cancellableTask.ensureNotCancelled();
```

Comment:
Is this needed? The transport layer will have just checked for cancellation before calling `messageReceived` I think.

Reply:
@DaveCTurner I am unfamiliar where this code is called within the transport layer. 

I copied this logic loosely based on NodeTransportHandler.

https://github.com/elastic/elasticsearch/blob/2f5ec518f349db7e446c0d4696666284bd983357/server/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java#L327-L335

Reply:
I will happily remove it. The big change I need for subsequent changes is that the task is cancellable, I thought it prudent to check here. But if its done for sure higher up the stack, then I am good with removing it

Reply:
> I copied this logic loosely based on NodeTransportHandler.

`git blame` --> guilty as charged 🤡 

Yeah this is not necessary, we're called from `RequestHandlerRegistry#processMessageReceived` which just a few lines earlier obtained its task from `TaskManager#register`, so called `TaskManager#registerCancellableTask`, and this method would have thrown if the task were already cancelled.

Reply:
@DaveCTurner I will remove it then :D thanks for the review!

Reply:
ok, removed